library/src/main/java/bio/terra/landingzone/common/utils/LandingZoneFlightBeanBag.java

```diff
@@ -103,4 +105,12 @@ public AzureCustomerUsageConfiguration getAzureCustomerUsageConfiguration() {
   public AzureCredentialsProvider getAzureCredentialsProvider() {
     return azureCredentialsProvider;
   }
+
+  public void setArmManagers(ArmManagers armManagers) {
```

Comment:
ArmManagers should be initialized for each flight separately. I think semantically this is not the right place to keep it here. This is more for service-like beans which are long lived. We can likely initialize it the same way as it was before but using the main flight class instead. In this case we can eliminate corresponding step too. From my point of view the only one disadvantage of this is that we need to pass ArmManagers into each step.
What are the advantages of such approach?

Reply:
Passing in the `ArmManagers` to each step in the constructor like we currently do doesn't work, because initialization requires information that is obtained via a step in the flight, so we can't initialize them at the start.

The normal way of passing information to later steps is the working map, but everything in that has to serialize to a string between steps (which `ArmManagers` does not).

There are a couple of reasons I landing on this method specifically:
- I assumed there was a cost to initializing `ArmManagers`, since it appears to need to connect to azure as part of the initialization process.
- This allows us to more easily pass mocks in for testing purposes. 
- While `ArmManagers` is different than other items in this class, insofar as it isn't instantiated automatically by spring, but it _is_ a complex object that is persisting across steps. When using the subflight approach, it was still functioning the same basic way (using a single instance that was shared across steps), so this was the least disruptive solution I could see.

The other practical solutions I see are:
1. Instantiating a new instance at each step: This is _possible_, because we have the information available at those future steps, and it's pretty easy from a code implementation standpoint (just share a method in the `BaseResourceCreateStep`. But even if there isn't a significant initialization cost, it creates difficulties in testing, since it becomes difficult to inject a mock version of the `ArmManagers`. It's possible using spying, but it requires additional setup steps in basically every test, exposing the method with higher visibility than it would otherwise need, and spying is less reliable than mocking, as a rule.
2.  Creating some sort of 'provider', and storing the provider in the `LandingZoneFlightBeanBag`: This seems like creating an overly complex bit of indirection for no other purpose than the sake of pedanticism and preserving a sense of aesthetics. 
3. We _could_ retrieve the billing profile _before_ the flight, which _would_ allow us to create the ArmManagers, and pass them in to each step like before. But we'd give up the automatic exception handling/retrying we get with stairway. Not the end of the world, but it seemed like this implementation was the lesser disruption.

Reply:
I think 3 might be a good option.

Reply:
That's what I went with.